### PR TITLE
Adding py-earth recipe

### PR DESCRIPTION
**Pull request by mehdidc**

### CODE REVIEW

**Comment by jakirkham** [comment]
Please rebase on current `master`. There is some clutter here that has since been removed.

File: recipes/py-earth/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+package:
+  name: pyearth
+  version: "0.1"
+
+source:
+    url: https://github.com/mehdidc/py-earth/archive/test_conda.zip
+    sha256: f3eca1b2e4ee408255433236df5f4d6c7854c5910214af91824a70eb17e41024
+
+build:
+    number: 0
```
Comment (jakirkham) [inline recipes/py-earth/meta.yaml]: Please add the following to `build`.

```
  script: python setup.py install --single-version-externally-managed --record record.txt
```

Then please drop `build.sh` and `bld.bat`.

File: recipes/py-earth/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+requirements:
+  build:
```
Comment (jakirkham) [inline recipes/py-earth/meta.yaml]: Please add `toolchain`.

File: recipes/py-earth/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    - python
+    - setuptools
+    - cython
+    - numpy
```
Comment (jakirkham) [inline recipes/py-earth/meta.yaml]: Please change to...

```
    - numpy x.x
```

File: recipes/py-earth/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    - scikit-learn
+    - nose
+    - six  
+  run:
+    - python
+    - numpy
```
Comment (jakirkham) [inline recipes/py-earth/meta.yaml]: Please change to...

```
    - numpy x.x
```

File: recipes/py-earth/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    - scipy
+    - scikit-learn
+    - nose
+  imports:
+    - pyearth 
+
+about:
+  home: https://github.com/scikit-learn-contrib/py-earth
+  summary : The py-earth package is a Pythom implementation of Jerome Friedman's Multivariate Adaptive Regression Splines algorithm (MARS) with a scikit-learn like interface.
+  license: BSD
```
Comment (jakirkham) [inline recipes/py-earth/meta.yaml]: Please specify number of clauses (e.g. `BSD 3-Clause`).

File: recipes/py-earth/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    - numpy
+    - scipy
+    - scikit-learn
+    - nose
+  imports:
+    - pyearth 
+
+about:
+  home: https://github.com/scikit-learn-contrib/py-earth
+  summary : The py-earth package is a Pythom implementation of Jerome Friedman's Multivariate Adaptive Regression Splines algorithm (MARS) with a scikit-learn like interface.
```
Comment (jakirkham) [inline recipes/py-earth/meta.yaml]: Could we please shorten the `summary`.

File: recipes/py-earth/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+    number: 0
+
+requirements:
+  build:
+    - python
+    - setuptools
+    - cython
+    - numpy
+    - scikit-learn
+    - nose
```
Comment (jakirkham) [inline recipes/py-earth/meta.yaml]: Is this needed for the build?

**Comment by jakirkham** [comment]
Thanks for your contribution, @mehdidc. Added some suggestions that should help this recipe build and clean it up a bit. We will need to update it with a proper release at some point, but am ok making sure it builds here before we do that.

**Comment by rth** [comment]
This was superseded by #5164 and can be closed now, I think.   cc @CJ-Wright . Thanks!

**Comment by jakirkham** [comment]
Guessing you meant ( https://github.com/conda-forge/staged-recipes/pull/5108 ). 😉

**Pull request closed without merging**